### ISSUE TITLE
[New Feature] Support for inline certificates and keys in Zenoh transport configuration.

### ISSUE THREAD

**Issue by gabrik** [eclipse-zenoh/zenoh#575]
### Describe the feature

It would be good to have the possibility to include the certificate and key data inline in the Zenoh configuration.
This allows for "portable" configuration files as they will not have any dependency on external files.

Nowadays if we want to include information about TLS/mTLS certificates we need to include the path, thus both configuration file and certificate files need to be shipped to the application. This can also cause errors if the path does not match between the development machine and the deployment machine.

Thus to have the possibility to inline them, ideally using `base64` encoding (similarly to what is done by Kubernetes in its configuration files.

An example configuration file will be:
```
{
    "connect": {
        "endpoints": [
            "tls/REDACTED.routers.REDACTED.zettascale.cloud:7447",
            "tls/REDACTED.routers.REDACTED.zettascale.cloud:7447"
        ]
    },
    "transport": {
        "link": {
            "tls": {
                "client_auth": true,
                "client_certificate_data": ["QkVHSU......"],
                "client_private_key_data": ["QkVHSA......"],
                "root_ca_certificate_data": ["QkVHSU......"],
                "server_name_verification": false
            }
        }
    }
}
```

I added them as string arrays because ideally, we could think to support multiple certificates/keys.

**Comment by gabrik** [comment]
In order to keep track of the work needed.

In order to support this the TLS configuration schema has to be updated: [zenoh-config/src/lib.rs](https://github.com/eclipse-zenoh/zenoh/blob/master/commons/zenoh-config/src/lib.rs#L288)

And then update the `ConfigurationInspector` of TLS (QUIC uses the same): [here](https://github.com/eclipse-zenoh/zenoh/blob/master/io/zenoh-links/zenoh-link-tls/src/lib.rs#L67).

The assumption is that only the path or only the data can be present, thus when inspecting the configuration an error must be returned if both are present.

**Comment by gabrik** [comment]
Solved by #577
